neo/SmartContract/InteropDescriptor.cs

```diff
@@ -11,25 +11,27 @@ internal class InteropDescriptor
         public long Price { get; }
         public Func<RandomAccessStack<StackItem>, long> PriceCalculator { get; }
         public TriggerType AllowedTriggers { get; }
+        public bool RequireWriteAccess { get; }
 
-        public InteropDescriptor(string method, Func<ApplicationEngine, bool> handler, long price, TriggerType allowedTriggers)
-            : this(method, handler, allowedTriggers)
+        public InteropDescriptor(string method, Func<ApplicationEngine, bool> handler, long price, TriggerType allowedTriggers, bool requireWriteAccess)
+            : this(method, handler, allowedTriggers, requireWriteAccess)
         {
             this.Price = price;
         }
 
-        public InteropDescriptor(string method, Func<ApplicationEngine, bool> handler, Func<RandomAccessStack<StackItem>, long> priceCalculator, TriggerType allowedTriggers)
-            : this(method, handler, allowedTriggers)
+        public InteropDescriptor(string method, Func<ApplicationEngine, bool> handler, Func<RandomAccessStack<StackItem>, long> priceCalculator, TriggerType allowedTriggers, bool requireWriteAccess)
+            : this(method, handler, allowedTriggers, requireWriteAccess)
         {
             this.PriceCalculator = priceCalculator;
         }
 
-        private InteropDescriptor(string method, Func<ApplicationEngine, bool> handler, TriggerType allowedTriggers)
+        private InteropDescriptor(string method, Func<ApplicationEngine, bool> handler, TriggerType allowedTriggers, bool requireWriteAccess)
         {
             this.Method = method;
             this.Hash = method.ToInteropMethodHash();
             this.Handler = handler;
             this.AllowedTriggers = allowedTriggers;
+            this.RequireWriteAccess = requireWriteAccess;
```

Comment:
I don't think this is a good idea @shargon... what is `RequireWriteAccess`? To Storage? I thought that, if current ExecutionContextState is running on ReadOnly, then methods that require write access could simply fail (example: Storage.GetCurrentContext`).

Reply:
We can rename the var, but not remove it, because is used for ban the syscall when is in readOnly mode